File: backend/models/pass_model.js
const mongoose = require("mongoose");
const schema = mongoose.Schema;

const PassSchema = new schema(
  {
    passId: {
      type: String,
      required: true,
    },
    uid: {
      type: String,
      required: true,
    },
    studentId: {
      type: String,
      required: true,
    },
    type: {
      type: String,
      required: true,
      emum: ["Gatepass", "Staypass"],
    },
    reason: {
      type: String,
      required: true,
    },
    status: {
      type: String,
      required: true,
      enum: ["Pending", "Approved", "Rejected", "Used", "In use", "Expired"],
    },
    destination: {
      type: String,
      required: true,
    },
    isActive: {
      type: Boolean,
      required: true,
    },
    approvedBy: {
      type: String,
      required: true,
      default: "None",
    },
    confirmedWith: {
      type: String,
      required: true,
      default: "None",
    },
    exitScanBy: {
      type: String,
    },
    exitScanAt: {
      type: String,
    },
    entryScanBy: {
      type: String,
    },
    entryScanAt: {
      type: String,
    },
    expectedOut: {
      type: String,
      required: true,
    },
    expectedIn: {
      type: String,
      required: true,
    },
    // actualOut: {
    //   type: String,
    // },
    // actualIn: {
    //   type: String,
    // },
    isEntryLate: {
      type: Boolean,
    },
    lateReason: {
      type: String,
    },
    qrId: {
      type: String,
      required: true,
    },
    isSpecialPass: {
      type: Boolean,
      required: true,
    },
  },
  { timestamps: true }
);

module.exports = mongoose.model("passes", PassSchema);

File: frontend/lib/models/pass_model.dart
class Pass {
  Pass(
      {required this.passId,
      required this.studentId,
      required this.gender,
      required this.qrId,
      required this.status,
      required this.destination,
      required this.type,
      required this.isActive,
      required this.reason,
      required this.expectedInTime,
      required this.expectedInDate,
      required this.expectedOutTime,
      required this.expectedOutDate,
      required this.isSpecialPass,
      this.actualInTime,
      this.actualInDate,
      this.actualOutTime,
      this.actualOutDate,
      this.showQr,
      this.isLate,
      this.isExceeding
      // this.actualInTime,
      // this.actualInDate,
      // this.actualOutTime,
      // this.actualOutDate,
      });

  final String passId;
  final String studentId;
  final String gender;
  final String qrId;
  final String status;
  final String destination;
  final String type;
  final bool isActive;
  final String reason;
  final String expectedInTime;
  final String expectedInDate;
  final String expectedOutTime;
  final String expectedOutDate;
  final String? actualInTime;
  final String? actualInDate;
  final String? actualOutTime;
  final String? actualOutDate;
  final bool isSpecialPass;
  bool? showQr;
  bool? isLate;
  bool? isExceeding;
  // final String? actualInTime;
  // final String? actualInDate;
  // final String? actualOutTime;
  // final String? actualOutDate;
}
